src/Service.Tests/Unittests/MultiSourceQueryExecutionUnitTests.cs

```diff
@@ -215,5 +280,56 @@ public async Task TestMultiSourceTokenSet()
             await msSqlQueryExecutor.SetManagedIdentityAccessTokenIfAnyAsync(conn, string.Empty);
             Assert.AreEqual(expected: db1AccessToken, actual: conn.AccessToken, "Data source connection failed to be set with default access token when source name provided is empty.");
         }
+
+        private static RuntimeConfig GenerateMockRuntimeConfigForMultiDbScenario()
+        {
+            // Set up mock config where we have two entities both mapping to different data sources.
+            Dictionary<string, Entity> entities = new()
+            {
+                { ENTITYNAME1, GraphQLTestHelpers.GenerateEmptyEntity() },
+                { ENTITYNAME2, GraphQLTestHelpers.GenerateEmptyEntity() }
+            };
+
+            Dictionary<string, DataSource> dataSourceNameToDataSource = new()
+            {
+                { DATASOURCENAME1, new(DatabaseType.MySQL, "Server =<>;Database=<>;User=xyz;Password=xxx", new())},
+                { DATASOURCENAME2, new(DatabaseType.CosmosDB_NoSQL, "Server =<>;Database=<>;User=xyz;Password=xxx", new()) }
+            };
+
+            Dictionary<string, string> entityNameToDataSourceName = new()
+            {
+                { ENTITYNAME1, DATASOURCENAME1 },
+                { ENTITYNAME2, DATASOURCENAME2 }
+            };
+
+            RuntimeConfig mockConfig1 = new(
+               Schema: "",
+               DataSource: new(DatabaseType.MySQL, "Server =<>;Database=<>;User=xyz;Password=xxx", new()),
+               Runtime: new(
```

Comment:
maybe i'm misremembering how we configure multi db scenario: don't we need two runtime config files? one for each source?

Reply:
We do but it all merges into one runtime config eventually. We merge the entities from both datasources into a single dictionary and take the runtime settings from the base config. Hence in test case, I directly just set the resultant runtimeconfig.